src/MiniProfiler.Shared/Timing.cs

```diff
@@ -77,28 +78,76 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public List<Timing> Children
+        {
+            get
+            {
+                lock(_lockObject)
+                {
+                    return _children == null ? null : new List<Timing>(_children);
+                }
+            }
+            set
+            {
+                lock(_lockObject)
+                {
+                    _children = value == null ? null : new List<Timing>(value);
+                }
+            }
+        }
 
         /// <summary>
         /// <see cref="CustomTiming"/> lists keyed by their type, e.g. "sql", "memcache", "redis", "http".
         /// </summary>
         [DataMember(Order = 6)]
-        public Dictionary<string, List<CustomTiming>> CustomTimings { get; set; }
+        public Dictionary<string, List<CustomTiming>> CustomTimings
```

Comment:
I don't think locks add much overhead actually. Most of the time it would be uncontested locking where the overhead is fairly minimal (this is an example i found someone attempted to measure it: http://kynosarges.org/LockPerformance.html) and when the locks are contested clearly we want to have them. 

Also I don't think there is any alternative to using locks. As soon as the code starts using DefaultProfileProvider (from the v4 branch) there is a possibility of multiple threads accessing single Timing, the only thing the application needs to do is to send requests to a couple of data sources (e.g. redis and database) asynchronously and wait for them all at one place (like using Task.WhenAll). I'm not sure how you envision to support mitigate this without using locks.